core/src/test/resources/sql/agg.iq

```diff
@@ -2990,4 +2990,146 @@ from (select 1 as a, 2 as b, 3 as c);
 
 !ok
 
+# [CALCITE-3661] Add MODE aggregate function
+
+# mode function applied to input value and return the most frequent value.
+select MODE(gender)
+from emp;
++--------+
+| EXPR$0 |
++--------+
+| F      |
++--------+
+(1 row)
+
+!ok
+
+# mode function applied to distinct value and return the first input value.
+select MODE(distinct gender)
+from emp;
++--------+
+| EXPR$0 |
++--------+
+| F      |
++--------+
+(1 row)
+
+!ok
+
+# mode function applied to filter and input value is not NULL.
+select MODE(gender)
+from emp
+where deptno <= 20;
++--------+
+| EXPR$0 |
++--------+
+| M      |
++--------+
+(1 row)
+
+!ok
+
+# mode function applied to filter and input value is NULL.
+select MODE(gender)
+from emp
+where deptno > 60;
++--------+
+| EXPR$0 |
++--------+
+|        |
++--------+
```

Comment:
Interesting. So if the input is only NULL so nothing will be returned?

Reply:
Yes. This result keeps consistent with PostgreSQL.

Reply:
@amaliujia If you have any other advice,  please let me know. Then we can make some progress.